### PR TITLE
c(7): Add C23 standard update

### PR DESCRIPTION
**Pull request by kfv**
This PR updates the `c(7)` man page with details about the **C23** standard, which was officially published on October 31, 2024. This update includes a summary of the significant changes in C23, such as new keywords, attribute additions, function enhancements, and deprecated features, along with context on its development timeline.

Please note that this is primarily focused on adding information for the recent C23 release. The overall structure and macro usage in the man page require revision to improve clarity, consistency, and adherence to formatting best practices, I might admit. I will address these improvements in a separate PR in the near future to enhance the man page structure and readability.

### CODE REVIEW

File: share/man/man7/c.7 [hunk 1]
```diff
@@ -55,7 +56,7 @@ different architectures.
 It follows that C is very eligible for (but not limited to) writing
 operating systems and low-level applications.
 .Pp
-The C language did not have a specification or standardized version for
+The C language did not have a specification or standardised version for
```
Comment (brooksdavis) [inline share/man/man7/c.7:59]: This isn't related to the commit message and is an en_US -> en_GB change.

File: share/man/man7/c.7 [hunk 2]
```diff
@@ -211,17 +212,120 @@ It incorporates the Technical Corrigendum 1 (ISO/IEC 9899:2011/COR1:2012)
...
+Elevate true and false to proper keywords (previously macros from
+<stdbool.h>)
+.It
+Add keywords alignas, alignof, bool, static_assert, thread_local; previously
+defined keywords remain available as alternative spellings
+.It
+Enable zero initialisation with {} (including initialisation of VLAs)
+.It
+Introduce C++11 style attributes using [[]], and support [[deprecated]],
+[[fallthrough]], [[maybe_unused]], [[nodiscard]], and [[noreturn]] for
```
Comment (brooksdavis) [inline share/man/man7/c.7:309]: Feels like a run on sentence from "for" onward.

File: share/man/man7/c.7 [hunk 3]
```diff
@@ -211,17 +212,120 @@ It incorporates the Technical Corrigendum 1 (ISO/IEC 9899:2011/COR1:2012)
...
+Remove trigraph support
+.It
+Remove K&R function definitions and declarations
+.It
+Remove non-two's-complement representations for signed integers
+.El
 .Pp
-Some useful features have been provided as extensions by some compilers, but
-they cannot be considered as standard features.
+The next version of the C Standard, informally named C2Y, is anticipated
```
Comment (brooksdavis) [inline share/man/man7/c.7:323]: "C2x" used above. Pick one case?

File: share/man/man7/c.7 [hunk 4]
```diff
@@ -349,15 +453,17 @@ C Standard.
...
 This manual page was originally written by
-.An Gabor Kovesdan Aq Mt gabor@FreeBSD.org .
-It was updated for
-.Fx 14.0
-by
-.An Faraz Vahedi Aq Mt kfv@kfv.io
-with information about more recent C standards.
+.An Gabor Kovesdan Aq Mt gabor@FreeBSD.org
+and is now maintained by
+.An Faraz Vahedi Aq Mt kfv@kfv.io .
```
Comment (brooksdavis) [inline share/man/man7/c.7:469]: We don't generally claim maintainership in the documentation.

File: share/man/man7/c.7 (hunk 1 above)
Comment (kfv) [reply share/man/man7/c.7:59]: Right, I'll split it into two commits.

File: share/man/man7/c.7 (hunk 2 above)
Comment (kfv) [reply share/man/man7/c.7:309]: Right.

File: share/man/man7/c.7 (hunk 3 above)
Comment (kfv) [reply share/man/man7/c.7:323]: Sorry, I didn't quite catch.

File: share/man/man7/c.7 (hunk 4 above)
Comment (kfv) [reply share/man/man7/c.7:469]: Makes sense, I'll revert it then, alright.

File: share/man/man7/c.7 (hunk 3 above)
Comment (brooksdavis) [reply share/man/man7/c.7:321]: C2x vs C2X, pick upper or lower for all uses.

Comment (kfv) [reply share/man/man7/c.7:321]: Oh, I see! Silly me, LOL.

File: share/man/man7/c.7 (hunk 1 above)
Comment (brooksdavis) [reply share/man/man7/c.7:59]: The project standard is American English https://docs.freebsd.org/en/books/fdp-primer/book/#writing-style-guidelines

Comment (kfv) [reply share/man/man7/c.7:59]: OK, one moment, please.

**Review by brooksdavis — APPROVED** [review]
Looks good now, I'll let it sit over the weekend in case anyone else wants to weigh in and aim to commit it Monday if not.

File: share/man/man7/Makefile [hunk 5]
```diff
@@ -42,7 +42,8 @@ MLINKS+= c.7 c95.7
 MLINKS+= c.7 c99.7
 MLINKS+= c.7 c11.7
 MLINKS+= c.7 c17.7
-MLINKS+= c.7 c2x.7
```
Comment (emaste) [inline share/man/man7/Makefile:45]: We'll need an ObsoleteFiles.inc entry to remove the old c2x.7

Comment (kfv) [reply share/man/man7/Makefile:45]: Thanks! I just pushed a commit, could you please take a look?

File: share/man/man7/c.7 [hunk 6]
```diff
@@ -211,17 +212,118 @@ It incorporates the Technical Corrigendum 1 (ISO/IEC 9899:2011/COR1:2012)
...
+.It
+Add null pointer type nullptr_t and the nullptr keyword
+.It
+Add constexpr keyword as a storage-class specifier for objects
+.It
+Redefine the usage of the auto keyword to support type inference while also
+retaining its previous functionality as a storage-class specifier when used
+with a type
+.It
+Add %b binary conversion specifier to the printf() and scanf() function
```
Comment (gperciva) [inline share/man/man7/c.7:235]: To nitpick, `mandoc -T lint` gives STYLE notices for all these function names without `.Fn`.

Normally I wouldn't bother mentioning style issues, but these are the only issues with `c.7`; previously, there were no issues at all with this file.

**Comment by gperciva** [comment]
Thanks!  I think this change is important enough to merit bumping the `.Dd` date on line 25:
https://github.com/freebsd/freebsd-src/pull/1507/files#diff-4f083ce9c5d62d6521e8da91acbc11ceaf6a30b28dee7a9cfdaa8e1ff92e2c14R25

File: share/man/man7/c.7 (hunk 6 above)
Comment (kfv) [reply share/man/man7/c.7:235]: Right, that’s why I mentioned the focus herein is only on adding information for the recent C23 release and the overall structure and macro usage in the man page shall be improved (and not just the function names—there’s quite a bit that needs proper macro usage, I might add.) Landing the update first would be ideal, and I can work on the style next week in a separate PR.

Comment (kfv) [reply share/man/man7/c.7:235]: Forget it, I'll do the initial revision in an hour.

File: ObsoleteFiles.inc [hunk 7]
```diff
@@ -51,6 +51,9 @@
 #   xargs -n1 | sort | uniq -d;
 # done
 
+# 20241102: remove the outdated c2x.7 in favor of c23.7
```
Comment (emaste) [inline ObsoleteFiles.inc:54]: I would be explicit that this is the c2x.7 symlink, not a separate page

Comment (kfv) [reply ObsoleteFiles.inc:54]: Alright, will do it now.

Comment (kfv) [reply ObsoleteFiles.inc]: Done.

**Review by gperciva — APPROVED** [review]
Looks good, thanks!

(Note that I'm a new doc contributor, not an established committer or maintainer, so my approval means nothing.  :)

**Comment by brooksdavis** [comment]
I pealed off the bits that fixed existing markup and squashed the rest of the commits. Also bumped date's to today.

Merged as of 930654318ecee172e4fc1ce57f21b4fb7b12603e

**Pull request closed without merging**